client.go

```diff
@@ -185,18 +188,36 @@ func isEmpty(a any) bool {
 	return v.IsZero() || v.Kind() == reflect.Ptr && v.IsNil()
 }
 
-func fmtQuery(query ...[2]string) string {
+func fmtQuery(operationID, method string, query ...[2]string) string {
 	q := make(url.Values)
+
+	// Add all provided query parameters
 	for _, v := range query {
-		q.Add(v[0], v[1])
+		if v[0] != "" { // Skip empty keys
+			q.Add(v[0], v[1])
+		}
 	}
 
-	if !q.Has("limit") {
+	// Add default limit for GET requests if conditions are met
+	const defaultLimit = "999"
+	if shouldAddDefaultLimit(operationID, method, q) {
 		// TODO: BAD hack to get around pagination in most cases
 		// we should implement this properly at some point but for now
 		// that should be its own issue
-		q.Add("limit", "999")
+		q.Add("limit", defaultLimit)
 	}
 
 	return q.Encode()
 }
+
+// shouldAddDefaultLimit determines if default limit should be added
+func shouldAddDefaultLimit(operationID, method string, q url.Values) bool {
+	var operationsWithoutLimit = []string{
+		"ServiceKafkaQuotaDescribe",
+		"ServiceKafkaQuotaDelete",
+	}
+
+	return !q.Has("limit") &&
+		method == http.MethodGet &&
+		!slices.Contains(operationsWithoutLimit, operationID)
+}
```

Comment:
Some methods might use POST to list items. So I would got with operation id filtering only.

nit: I believe switch would do the job too. We should eventually remove this hack and explicitly add the "limit" parameters where they belong.

Reply:
upd